src/main/java/org/kairosdb/datastore/cassandra/CassandraDatastore.java

```diff
@@ -656,69 +661,165 @@ private static void filterAndAddKeys(String metricName, SetMultimap<String, Stri
         }
     }
 
+    // TODO remove when old getMatchingRowKeys is uncommented
     private List<DataPointsRowKey> getMatchingRowKeys(String metricName, long startTime, long endTime, SetMultimap<String, String> filterTags, int limit) {
-        final List<DataPointsRowKey> rowKeys = new ArrayList<>();
-        final DataPointsRowKeySerializer keySerializer = new DataPointsRowKeySerializer();
-
+        // determine whether to use split index or not
+        final List<DataPointsRowKey> rowKeys = new LinkedList<>();
```

Comment:
Please provide your opinion on this `LinkedList`. I noticed that operations over this list are all either writes or sequential reads, but never seeks. Thanks why `LinkedList` and not `ArrayList`.